streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -549,12 +540,12 @@ void to(final TopicNameExtractor<K, V> topicExtractor,
      *     }
      * }
      * }</pre>
+     * Even if any upstream operation was key-changing, no auto-repartition is triggered.
+     * If repartitioning is required, a call to {@link #through(String)} should be performed before {@code transform()}.
```

Comment:
It's a bummer, since it forces you to come up with a name for the repartition topic, which also makes its lifecycle your responsibility. And also, Streams won't know that it's "just" a repartition topic, so it can't be optimized.

Can I create a Jira to add a way to mark the transform as "repartition required"? Or maybe we can automatically do this when the transform has a state store?

Reply:
There is already a JIRA: https://issues.apache.org/jira/browse/KAFKA-7608 and also inactive KIP https://cwiki.apache.org/confluence/display/KAFKA/KIP-221%3A+Enhance+KStream+with+Connecting+Topic+Creation+and+Repartition+Hint

It's not fully backed out yet, but there is stuff we could do to improve the user experience.

Reply:
:+1: Thanks!